docs/README.md

```diff
@@ -0,0 +1,613 @@
+
+### HTTP
+
+Apso provides a tiny wrapper for [Dispatch](http://dispatch.databinder.net/) with synchronous operations. It's called `W`, and the following shows some sample usage:
+
+```scala mdoc:compile-only
+import com.velocidi.apso.http.W
+
+W.get("http://www.google.com/").getStatus
+// res0: Int = 302
+
+W.post("http://www.google.com/", "").getStatus
+// res1: Int = 405
+
+W.put("http://www.google.com/", "").getStatus
+// res2: Int = 405
+
+W.delete("http://www.google.com/").getStatus
+// res3: Int = 405
+
+W.head("http://www.google.com/").getStatus
+// res4: Int = 302
+```
+
+The POST and PUT methods can also receive the body as `JSON` (of [circe](https://github.com/circe/circe)), which adds the `Content-type` header accordingly.
+
+### Geo
+
+The `Geo` object provides methods to compute distances in kilometers between two points on the planet Earth, calculated using the spherical [law of cosines](https://en.wikipedia.org/wiki/Great-circle_distance#Formulas). Coordinates are represented by a pair of `Double` for latitude and longitude.
+
+```scala mdoc:reset
+import com.velocidi.apso.Geo
+
+Geo.distance((41.1617609, -8.6024716), (41.1763745, -8.5964861))
+```
+
+You can also have the distance function curried if you are computing distances from a fixed point:
+
+```scala mdoc
+val distFromOffice = Geo.distanceFrom((41.1617609, -8.6024716))
+
+distFromOffice((41.1763745, -8.5964861))
+
+distFromOffice((38.7223032, -9.1414664))
+```
+
+### Implicits
+
+Apso provides implicit conversions from `String`, `Seq[_]`, `Map[_, _]`, `Seq[Map[_, _]]` and `AutoCloseable` to extended types that come packed with extended features.
+
+```scala mdoc:reset
+import com.velocidi.apso.Implicits._
+
+Seq(1, 3, 5).mergeSorted(Seq(2, 4))
+
+(0 to 15).average
+
+Map(1 -> 2, 3 -> 6).twoWayMerge(Map(2 -> 4, 3 -> 5)) { (a, b) => b }
+
+Map(1 -> 2, 2 -> 4, 3 -> 6).twoWayMerge(Map(2 -> 2, 3 -> 5)) { (a, b) => b }
+
+Map(1 -> 2, 2 -> 3).mapKeys(_ + 1)
+
+scala.util.Random.choose((0 to 15).toSeq)
+
+scala.util.Random.choose((0 to 15).toSeq)
+
+scala.util.Random.choose((0 to 15).toSeq)
+
+scala.util.Random.choose((0 to 15).toSeq)
+
+scala.util.Random.chooseN((0 to 15).toSeq, 4)
+
+scala.util.Random.chooseN((0 to 15).toSeq, 4)
+```
+
+### JreVersionHelper
+
+The JreVersionHelper object provides helper methods to check the two most significant parts of the JRE version at runtime:
+
+```scala mdoc:reset
```

Comment:
This should also probably be behind a `compile-only` block, since it depends on the platform the docs are compiled in.